// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

import "memory_plugin_common.proto";

// Memory plug-in configuration, passed to plug-in by plug-in service.
message MemoryConfig {
    // set true to report process list
    bool report_process_tree = 1;
    // set true to report memory counter from /proc/meminfo
    bool report_sysmem_mem_info = 2;
    // set required counter list of system meminfo, eg:MemTotal, MemFree, etc.
    repeated SysMeminfoType sys_meminfo_counters = 3;
    // set true to report memory counter from /proc/vmstat
    bool report_sysmem_vmem_info = 4;
    // set required counter list of virtual system meminfo, eg:nr_free_pages, nr_anon_pages, etc.
    repeated SysVMeminfoType sys_vmeminfo_counters = 5;
    // set true to report process meminfo from /proc/${pid}/stat
    bool report_process_mem_info = 6;
    // set true to report application memory usage summary, eg:java heap memory, native heap, stack memory, etc.
    bool report_app_mem_info = 7;
    // set true to report application memory by dumpsys service, otherwise,
    // application memory will count up by /proc/${pid}/smaps information
    bool report_app_mem_by_dumpsys = 8;
    // set required pid list
    repeated int32 pid = 9;
}
